zerver/tests/test_webhooks_common.py

```diff
@@ -117,3 +120,47 @@ def test_missing_event_header(self) -> None:
 
     def get_body(self, fixture_name: str) -> str:
         return self.webhook_fixture_data("groove", fixture_name, file_type="json")
+
+class TestHttpApiCallback(ZulipTestCase):
+
+    @patch('requests.get')
+    def test_http_api_callback_with_simple_token_auth_strategy(self,
+                                                               requests_get_mock: MagicMock
+                                                               ) -> None:
+        webhook_bot = get_user("webhook-bot@zulip.com", get_realm("zulip"))
+        url = "https://example.com/api/v2/endpoint"
+        # Test to see if the right error would be raised for an unsupported auth strategy.
+        with self.assertRaises(ValueError):
+            http_api_callback(webhook_bot, 9999, url)
+
+        # Now test to see if and how the method complains whenever any required
+        # BotConfigData if not present.
+        msg = {"message": "Successful!"}
+        requests_get_mock.return_value = SimpleNamespace(status_code = 200,
+                                                         text = json.dumps(msg))
+
+        response = http_api_callback(webhook_bot, AuthStrategies.SIMPLE_TOKEN, url)
+        self.assertIsNone(response)
+
+        BotConfigData.objects.create(bot_profile=webhook_bot,
+                                     key="callback_auth_param",
+                                     value="token")
+        response = http_api_callback(webhook_bot, AuthStrategies.SIMPLE_TOKEN, url)
+        self.assertIsNone(response)
+
+        # Test for success.
+        BotConfigData.objects.create(bot_profile=webhook_bot,
+                                     key="callback_auth_token",
+                                     value="s3creTkey")
+        response = http_api_callback(webhook_bot, AuthStrategies.SIMPLE_TOKEN, url)
+        self.assertEqual(response, {"message": "Successful!"})
+
+        # See if the method raises a UnsuccessfulResponseError for a non-200 response.
+        msg = {"message": "Failure!"}
+        requests_get_mock.return_value = SimpleNamespace(status_code = 403, text = json.dumps(msg))
+        try:
+            http_api_callback(webhook_bot, AuthStrategies.SIMPLE_TOKEN, url)
+        except UnsuccessfulResponseError as err:
+            expected_message = "API Callback to https://example.com/api/v2/endpoint failed with \
+status 403. Please check the configuration of the \"Zulip Webhook Bot\" bot."
+            self.assertEqual(str(err), expected_message)
```

Comment:
This would be much easier to review if you could please split this up into multiple tests with descriptive names! :)

Reply:
Agreed, I'll definitely make the change.